src/ngraph_builder1.h

```diff
@@ -0,0 +1,138 @@
+ *******************************************************************************/
+#ifndef NGRAPH_TF_BRIDGE_BUILDER1_H_
+#define NGRAPH_TF_BRIDGE_BUILDER1_H_
+
+#include <ostream>
+#include <vector>
+
+#include "ngraph/ngraph.hpp"
+#include "ngraph_log.h"
+#include "ngraph_mark_for_clustering.h"
+
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/graph/algorithm.h"
+
+using namespace std;
+namespace ng = ngraph;
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+// TODO (sarkars): make sure all comments from old builder are copied correctly.
+
+// The purpose of Builder is to accept a TF graph and convert it to a
+// equivalent ngraph fucntion. To that end, it acts as an interface to
+// a library of 'TranslateOps' functions. These functions are defined
+// in ngraph_translateops.h. Builder just goes through each TF op and
+// calls the appropriate 'TranslatorFn'. It does not implement the per-op
+// translation, the heavy-lifting being done by ngraph_translateops.h;
+// It merely does some bookkeeping, such as process the nodes in correct
+// order, keep track of and retrieve parent nGraph nodes etc.
+
+class Builder1 {
+ public:
+  using TranslatorFn = function<Status(
+      const Node*, vector<shared_ptr<ng::Node>>&,
+      const std::vector<const Tensor*>&, vector<shared_ptr<ng::Node>>&)>;
+
+  // Note: in case we want to get rid of Initialize, pass an
+  // OpKernelConstruction* ctx to the constructor, and use OP_REQUIRES_OK
+  // instead of TF_RETURN_IF_ERROR. This gets rid of Initialize() as it
+  // can handle errors in construction.
+  // But in the case of overloaded constructor that does not accept a ctx,
+  // which is used for OpExecuter test class, we cannot handle error during
+  // construction.
+  // Hence keeping the Initialize() function
+
+  // This constructor is for actual use (encapsulate_op)
+  Builder1(const Graph& tf_graph, OpKernelConstruction* ctx)
+      : m_tf_graph(tf_graph), m_is_initialized(false) {}
+
+  // And this version is for tests (OpExecuter)
+  Builder1(const Graph& tf_graph) : Builder1(tf_graph, nullptr) {}
+
+  // TranslateGraph is overloaded. This is for actual use (encapsulate_op)
+  Status TranslateGraph(OpKernelContext* ctx,
+                        std::shared_ptr<ngraph::Function>& ng_function);
+
+  // And this version is for tests (OpExecuter)
+  Status TranslateGraph(const std::vector<TensorShape>&,
+                        const std::vector<const Tensor*>&,
+                        shared_ptr<ng::Function>&);
+
+ private:
+  // The op map holds a mapping from TensorFlow op names (strings) to
+  // vector of generated nGraph nodes. Since we process nodes in toposort
+  // order, it is guaranteed when processing a TF node, its parents
+  // will already have been processed and safely stowed in ng_op_map
+  std::unordered_map<std::string, vector<shared_ptr<ng::Node>>> m_ng_op_map;
```

Comment:
Google Style Guide suggests adding trailing _ to all private data members of a class. nGraph core follows the style adopted here. 
https://google.github.io/styleguide/cppguide.html#Variable_Names
@avijit-nervana which one is recommended? Actually we can do both, by adding _ to all the variables.

Reply:
The m_ means data member, and s_ means static members. According to Adam, ngraph follows m_ and s_, so I have that format. Not sure if we stick to (m_, s_) or trailing _

Reply:
@sayantan-nervana When you get a chance - please resolve the conflict.